### DISCUSSION TITLE
Instalación Python para Notebooks

### DISCUSSION THREAD

**Discussion by LuceroLuciano** [LaunchX-InnovaccionVirtual/CursoIntroPython#48] [Q&A] [answered]
# Hola amigxs explorers! 

### Necesito de su ayuda 🙏

Estoy tratado de hacer la kata numero 0 desde el día domingo, pero nomas no puedo con la ejecución de los notebooks, Tengo la versión de  python 3.6 por defecto, pero instale la versiòn 3.9, solo que para ejecutar el código python en el notebook me aparece algo como esto:
![Captura de pantalla de 2022-02-09 11-42-32](https://user-images.githubusercontent.com/83784155/153261897-fe4336fa-583b-46d0-89b4-20acc9ee803f.png)

Con Markdown no tengo problema alguno, funciona bien pero el detalles esta con el código. También instale la versión 3.7 de python pero ahora no se como eliminar la versión 3.9 o utilizar la 3.7 por defecto, tambien tengo la duda de sabe si hay algún detalle que tenga instaladas muchas versiones de python en una misma computadora. Las siguiente captura de pantalla son todas las versiones que tengo instaladas: 

![Captura de pantalla de 2022-02-09 13-25-20](https://user-images.githubusercontent.com/83784155/153275336-01ff683b-12b7-4381-a2cf-ca43b391034c.png)
 
Estos son todos los kernel que me aparecen, no me aparece python 3.7 para probar la ejecución del código.
![Captura de pantalla de 2022-02-09 13-26-45](https://user-images.githubusercontent.com/83784155/153275701-29ea88cb-908a-4408-9804-ff54817eec7d.png)

Y cuando trato de ejecutar el código solo me aparece el mensaje para instalar ipykernel pakage:
![Captura de pantalla de 2022-02-09 11-42-18](https://user-images.githubusercontent.com/83784155/153276105-18ee99d0-1357-433d-9af4-48a051dee676.png)

Al darle click en instalar me aparece lo siguiente: 
![Captura de pantalla de 2022-02-09 13-31-48](https://user-images.githubusercontent.com/83784155/153276557-39eff61d-a1b2-4abf-8fa3-77d8aff23b9e.png)
![Captura de pantalla de 2022-02-09 13-32-21](https://user-images.githubusercontent.com/83784155/153276651-910822c5-e51d-4759-b402-ca1ccc51fa57.png)

Recuerdo que Investigando encontré algo que decía que la instalación del modulo pip estaba roto para python3.9, o eso fue lo que entendí. Como ultimo punto, estoy trabajando con Linux Ubuntu 18.04, si alguien me puede ayudar porfavor 🙏🙏🙏 se lo agradeceré infinitamente, voy algo atrazada con las katas y aún no queda mi entorno de trabajo.

**Comment by MarcosSilvaG** [comment]
intenta instalar manualmente el kernel en la terminal de vscode `pip install ipykernel`

**Reply by MarcosSilvaG** [reply]
si eso no funiona intenta usar otro[ entorno de python](https://code.visualstudio.com/docs/python/environments) o directamente reinstalar [python](https://www.python.org/)

**Reply by Ricardo-g-garcia** [reply]
a mi también me apareció el mismo mensaje y lo instale con pip install ipykernal y si me funciono

**Reply by bryan-vg** [reply]
hola, puedes intentar instalar el ipykernel en la terminal de vscode como recomienda marcos, o ya que dices que estas usando linux, puedes verificar si ubuntu ya viene con python y en caso que asi sea, instalar el ipykernel desde la terminal de linux con el comando:`pip install ipykernel` (claro que tambien debes tener instalado pip). En mi caso (uso debian) me salia el mismo error, luego instale el ipykernel directamente desde la terminal, volvi a abrir el vscode y ya no daban ese error las notebooks

**Reply by LuceroLuciano** [reply]
Ya instale lo de  `pip install ipykernel` pero ahora me dice que instale *notebook package*

![Captura de pantalla de 2022-02-10 13-55-31](https://user-images.githubusercontent.com/83784155/153490596-72a3b0e7-47cb-4e28-b225-5bc6f7027041.png)

Trato de instalar *notebook package*  desde la opción que dice instalar y también desde la terminal pero solo se instala "como una parte", por así decirlo, luego me marca algunas excepciones y/o errores. 

![Captura de pantalla de 2022-02-10 14-27-05](https://user-images.githubusercontent.com/83784155/153491003-8b36039f-68cf-471d-a4de-7a783ef4e37d.png)
![Captura de pantalla de 2022-02-10 14-27-19](https://user-images.githubusercontent.com/83784155/153491052-283a5584-e721-4b62-ac7d-4f3450c0df78.png)
![Captura de pantalla de 2022-02-10 14-27-28](https://user-images.githubusercontent.com/83784155/153491067-bf3a7c30-2735-417d-b7c3-dfbb9e260a82.png)

**Reply by bryan-vg** [reply]
una duda, si tienes instalada en el vs code la extension de jupyter? Y si asi es, ya que te dio error el comando que estas intentado usar podrias checar este link, espero te ayude: [https://medium.com/analytics-vidhya/how-to-install-jupyter-notebook-using-pip-e597b5038bb1](url)

**Comment by 3rickDJ** [comment, marked as answer]
# Directo desde terminal :keyboard: 


Podrías intentar instalar los paquetes directamente desde tu terminal, para que utilices el instalador pip que tienes por defecto; y no el que viene con python3.9 (como lo hace VScode), es recomendable que actualices tu `pip` con el siguiente comando (en la terminal):

`pip install --upgrade pip`

y luego instalar los paquetes desde tu terminal:

`pip install ipykernel`

### De hecho ipykernel depende a su vez de otros paquetes, este comando instalará los paquetes necesarios para poder utilizar jupyter notebooks. (supongo que al instalar ipykernel se instaló mal por utilizar el módulo pip de python3.9, por eso te recomiendo utilizar el pip por sí solo y actulizarlo)

**Déjame saber si te ayudó o no**  :mage_woman:

**Comment by LuceroLuciano** [comment]
Muchas gracias por su ayuda explorers. No logre solucionar el detalle en linux, así que decidí trabajar con Windows y ahora todo fluye mejor!
